src/Core/Managed/Shared/Channel/InMemoryTransmitter.cs

```diff
@@ -80,7 +78,19 @@ public void Dispose()
         /// </summary>
         internal void Flush(TimeSpan timeout)
         {
-            this.DequeueAndSend(timeout);
+#if !CORE_PCL
+            SdkInternalOperationsMonitor.Enter();
+            try
+            {
+#endif
+                this.DequeueAndSend(timeout);
```

Comment:
Can you instead do Enter/Exit in the body of DequeueAndSend() so Flush() and Runner() need not separately be marked with Enter/Exit.